File: src/modules/PosicoesVeiculos/typeorm/entities/PosicaoVeiculo.ts
import Veiculo from '@modules/veiculos/typeorm/entities/Veiculo';
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  OneToOne,
  PrimaryGeneratedColumn,
} from 'typeorm';

@Entity('posicaoVeiculos')
class PosicaoVeiculo {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  latitude: number;

  @Column()
  longitude: number;

  @OneToOne(() => Veiculo)
  @JoinColumn({ name: 'veiculoId' })
  veiculo: Veiculo;

  @CreateDateColumn()
  created_at: Date;

  @CreateDateColumn()
  updated_at: Date;
}

export default PosicaoVeiculo;

File: TransportePublico/TransportePublico.Domain/PosicaoVeiculo.cs
namespace TransportePublico.Domain
{
    public class PosicaoVeiculo
    {
        public long Id { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public long VeiculoId { get; set; }
    }
}
